python/ray/data/_internal/util.py

```diff
@@ -535,6 +536,35 @@ def validate_compute(
             )
 
 
+def get_compute_strategy(
+    compute: Optional[Union[str, "ComputeStrategy"]] = None,
+    concurrency: Optional[Union[int, Tuple[int, int]]] = None,
+) -> "ComputeStrategy":
+    # Lazily import these objects to avoid circular imports.
+    from ray.data._internal.compute import ActorPoolStrategy, TaskPoolStrategy
+
+    if compute is not None:
+        logger.warning(
+            "The argument `compute` is deprecated in Ray 2.9. Please specify "
+            "argument `concurrency` instead. For more information, see "
+            "https://docs.ray.io/en/master/data/transforming-data.html#"
+            "transforming-batches-with-actors."
+        )
+        return compute
+    else:
+        if isinstance(concurrency, tuple):
+            assert (
+                len(concurrency) == 2
+                and isinstance(concurrency[0], int)
+                and isinstance(concurrency[1], int)
+            )
+            return ActorPoolStrategy(min_size=concurrency[0], max_size=concurrency[1])
+        elif isinstance(concurrency, int):
+            return ActorPoolStrategy(size=concurrency)
```

Comment:
Do we allow actors if a non-callable function is passed?

Reply:
Before this PR, we allow it. Changed the code here to disallow actor if passing a function.